jsign-core/src/main/java/net/jsign/jca/HashiCorpVaultSigningService.java

```diff
@@ -0,0 +1,283 @@
+            }
+
+            /*
+             * HashiCorp Vault Google Cloud KMS key type format : 'rsa_sign_pkcs1_<BITS>_<DIGEST_ALGORITHM>'
+             * HashiCorp Vault Transit key type format          : 'rsa-<BITS>'
+             */
+            if (this.googleCloudKMS) {
+                algorithm = algorithm.substring(0, algorithm.indexOf("_")).toUpperCase();
+            }
+            else {
+                algorithm = algorithm.substring(0, algorithm.indexOf("-")).toUpperCase();
+            }
+
+        } catch (ClassCastException | IllegalArgumentException e) {
+            logger.log(Level.SEVERE, "Invalid response format while fetching algorithm for key alias '" + alias +"': ", e.getMessage());
+            throw (UnrecoverableKeyException) new UnrecoverableKeyException("Unable to fetch HashiCorp Vault private key '" + alias + "'").initCause(e);
+        } catch (IOException e) {
+            logger.log(Level.SEVERE, "IO Exception while fetching algorithm for key alias '" + alias + "': ", e.getMessage());
+            throw (UnrecoverableKeyException) new UnrecoverableKeyException("Unable to fetch HashiCorp Vault private key '" + alias + "'").initCause(e);
+        } catch (Exception e) {
+            logger.log(Level.SEVERE, "Unexpected exception while fetching algorithm for key alias '" + alias + "': ", e.getMessage());
+            throw (UnrecoverableKeyException) new UnrecoverableKeyException("Unexpected exception while fetching HashiCorp Vault private key '" + alias + "'").initCause(e);
+        }
+
+        SigningServicePrivateKey key = new SigningServicePrivateKey(alias, algorithm, this);
+        keys.put(alias, key);
+        return key;
+    }
+
+    @Override
+    public byte[] sign(SigningServicePrivateKey privateKey, String algorithm, byte[] data) throws GeneralSecurityException {
+        DigestAlgorithm digestAlgorithm = DigestAlgorithm.of(algorithm.substring(0, algorithm.toLowerCase().indexOf("with")));
+        data = digestAlgorithm.getMessageDigest().digest(data);
+
+        String alias = privateKey.getId();
+        String keyName = alias.substring(0, alias.indexOf(":"));
+        String keyVersion = alias.substring(alias.indexOf(":") + 1);
+
+        Map<String, Object> request = new HashMap<>();
+        request.put("key_version", keyVersion);
+
+        if (this.googleCloudKMS) {
+            request.put("digest", Base64.getEncoder().encodeToString(data));
+        }
+        else {
+            String shaType;
+            try {
+                shaType = digestAlgorithm.toString().split("SHA")[1];
+            } catch (RuntimeException e) {
+                logger.log(Level.WARNING, "Could not determine SHA type - defaulting to SHA256");
+                shaType = "256";
+            }
+
+            request.put("input", Base64.getEncoder().encodeToString(data));
+            request.put("prehashed", true);
+            request.put("hash_algorithm", String.format("sha2-%s", shaType));
```

Comment:
`String.format("sha2-%s", shaType)` is a bit overkill for a simple concatenation, `"sha2-" + shaType` is ok. Transit doesn't support SHA1?

Reply:
Makes sense, I've changed it accordingly in the next commit.
The formatting for SHA1 is a bit different from any SHA2 algorithm, as you can see in [HashiCorp Vault Official Documentation - Transit API - Sign Data](https://developer.hashicorp.com/vault/api-docs/secret/transit#sign-data).